### PR TITLE
Fix resource data(#38,#39)

### LINKED ISSUE #38: if website or phone isn't  there, just don't show that field
Command: `test dl` and scroll down to last data.
Screenshot from 2020-06-15 22-21-40

### LINKED ISSUE #39: `test tn` throws a error
```
Unhandled rejection Error: ETELEGRAM: 400 Bad Request: message is too long
    at request.then.resp (/home/gurrrung/fun/CovidBot/node_modules/node-telegram-bot-api/src/telegram.js:283:15)
    at tryCatcher (/home/gurrrung/fun/CovidBot/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/gurrrung/fun/CovidBot/node_modules/bluebird/js/release/promise.js:547:31)
    at Promise._settlePromise (/home/gurrrung/fun/CovidBot/node_modules/bluebird/js/release/promise.js:604:18)

```

### PR DESCRIPTION
**Pull request by baljinderxd**
closes #38 and closes #39 
for #39, resources will be shown in two separate messages to counter 
`Unhandled rejection Error: ETELEGRAM: 400 Bad Request: message is too long` if message length is greater than the limit.

### CODE REVIEW

**Comment by baljinderxd** [comment]
My bad I named this branch as `fix-test-data`.
Should have named it `fix-resource-data`.Hope the name do not creates any confusion 👍

File: app/utils/prepareAnswer.js [hunk 1]
```diff
@@ -126,19 +126,28 @@ function prepareDailyStatsAnswer(body, n) {
 
 function prepareTestingResourceAnswer(stateResources) {
   if (stateResources.length === 0) {
-    return '\n\n\u{1F9EA} No COVID-19 Testing Labs found';
+    const temp = ['\n\n\u{1F9EA} No COVID-19 Testing Labs found', ''];
```
Comment (gurrrung) [inline app/utils/prepareAnswer.js:129]: This was unnecessary.

File: app/commands/testingCentres.js [hunk 2]
```diff
@@ -33,9 +33,17 @@ function testingCentres(body, bot, chatId, stateUserMessage) {
 
       const data = prepareAnswer.prepareTestingResourceAnswer(stateResources);
 
-      return bot.sendMessage(chatId, data, {
+      return bot.sendMessage(chatId, data[0], {
         parse_mode: 'HTML',
         disable_web_page_preview: true,
+      }).then(() => {
```
Comment (gurrrung) [inline app/commands/testingCentres.js:39]: Let's assume the data is even longer and data[1] also throws the error?

File: app/utils/prepareAnswer.js [hunk 3]
```diff
@@ -126,19 +126,28 @@ function prepareDailyStatsAnswer(body, n) {
...
-
-  stateResources.forEach((stateResource) => {
-    data += `\n<b>${stateResource.nameoftheorganisation}, ${stateResource.city}\n`
-      + `</b>Description: ${stateResource.descriptionandorserviceprovided}.\n`
-      + `<i>Phone Number: ${stateResource.phonenumber}</i>\n`
-      + `<i>Website: ${stateResource.contact}</i>\n`;
-  });
-
-  return data;
+  let data2 = '';
```
Comment (gurrrung) [inline app/utils/prepareAnswer.js:134]: Use a loop to get all the data in an array.

File: app/commands/testingCentres.js [hunk 4]
```diff
@@ -33,9 +33,17 @@ function testingCentres(body, bot, chatId, stateUserMessage) {
 
       const data = prepareAnswer.prepareTestingResourceAnswer(stateResources);
 
-      return bot.sendMessage(chatId, data, {
+      return bot.sendMessage(chatId, data[0], {
```
Comment (gurrrung) [inline app/commands/testingCentres.js:36]: Use further a loop here for sending a message by fetching data from the array into small acceptable parts.
Do an empty return in the end.

**Comment by baljinderxd** [comment]
Stored data in form of array and it will display max 10 items in a single message. Data can be of any length now it would be divided in 10 items per message. Thanks for pointing out the problems @gurrrung 🤓

File: app/utils/prepareAnswer.js [hunk 5]
```diff
@@ -126,18 +126,22 @@ function prepareDailyStatsAnswer(body, n) {
...
-  let data = `\n\n\u{1F9EA} COVID-19 Testing Labs in ${stateResources[0].state}, found ${stateResources.length} results\n`;
-
-  stateResources.forEach((stateResource) => {
-    data += `\n<b>${stateResource.nameoftheorganisation}, ${stateResource.city}\n`
-      + `</b>Description: ${stateResource.descriptionandorserviceprovided}.\n`
-      + `<i>Phone Number: ${stateResource.phonenumber}</i>\n`
-      + `<i>Website: ${stateResource.contact}</i>\n`;
-  });
+  const data = [`\n\n\u{1F9EA} COVID-19 Testing Labs in ${stateResources[0].state}, found ${stateResources.length} results\n`];
+  data[0] += `\n<b>${stateResources[0].nameoftheorganisation}, ${stateResources[0].city}\n`
```
Comment (gurrrung) [inline app/utils/prepareAnswer.js:134]: Why can't we merge this with the `for loop`?

Comment (baljinderxd) [reply app/utils/prepareAnswer.js:134]: If we have to merge it in for loop, we will have to use `data[i]+=....` which will work good for the 1st element but for rest using `data[i]+=...` in very first line was producing undefined on top like this.
gur2

Comment (baljinderxd) [reply app/utils/prepareAnswer.js:134]: I am changing it to include in for loop with if else which will work similarly. 👍

File: app/utils/prepareAnswer.js [hunk 6]
```diff
@@ -126,18 +126,23 @@ function prepareDailyStatsAnswer(body, n) {
...
-    data += `\n<b>${stateResource.nameoftheorganisation}, ${stateResource.city}\n`
-      + `</b>Description: ${stateResource.descriptionandorserviceprovided}.\n`
-      + `<i>Phone Number: ${stateResource.phonenumber}</i>\n`
-      + `<i>Website: ${stateResource.contact}</i>\n`;
-  });
+  const data = [`\n\n\u{1F9EA} COVID-19 Testing Labs in ${stateResources[0].state}, found ${stateResources.length} results\n`];
 
+  for (let i = 0; i < stateResources.length; i += 1) {
+    if (i === 0) {
+      data[i] += `\n<b>${stateResources[i].nameoftheorganisation}, ${stateResources[i].city}\n`
```
Comment (gurrrung) [inline app/utils/prepareAnswer.js:137]: This loop can still be minimized, you don't even need if statement. Simply create a string of data and use `push` to push it to the end of the array.

Comment (baljinderxd) [reply app/utils/prepareAnswer.js:137]: Okay got it, will apply this change 😀

File: app/utils/prepareAnswer.js [hunk 7]
```diff
@@ -131,17 +131,14 @@ function prepareTestingResourceAnswer(stateResources) {
   }
 
   const data = [`\n\n\u{1F9EA} COVID-19 Testing Labs in ${stateResources[0].state}, found ${stateResources.length} results\n`];
+  let tempData;
```
Comment (gurrrung) [inline app/utils/prepareAnswer.js:134]: Please move this inside the loop and then we are good to go. Just directly use it like `let tempData = bla bla`.

Comment (baljinderxd) [reply app/utils/prepareAnswer.js:134]: okay doing it 👍

**Pull request merged**